pkg/controller/worker/machine_dependencies.go

```diff
@@ -1,13 +1,72 @@
 package worker
 
-import "context"
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	retryutils "github.com/gardener/gardener/pkg/utils/retry"
+	fcmv2 "github.com/metal-stack/firewall-controller-manager/api/v2"
+	"github.com/metal-stack/gardener-extension-provider-metal/pkg/metal"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
 // DeployMachineDependencies implements genericactuator.WorkerDelegate.
 func (w *workerDelegate) DeployMachineDependencies(_ context.Context) error {
 	return nil
 }
 
 // CleanupMachineDependencies implements genericactuator.WorkerDelegate.
-func (w *workerDelegate) CleanupMachineDependencies(_ context.Context) error {
-	return nil
+func (w *workerDelegate) CleanupMachineDependencies(ctx context.Context) error {
+	worker := w.worker.DeepCopy()
+	err := w.client.Get(ctx, client.ObjectKeyFromObject(w.worker), worker)
```

Comment:
Is this still necessary?

Reply:
I will try it again without re-fetching the worker resource.